src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -105,28 +111,32 @@ private WindowsIdentity()
         /// <summary>
         /// Initializes a new instance of the WindowsIdentity class for the user represented by the specified User Principal Name (UPN).
         /// </summary>
-        /// <remarks>
-        /// Unlike the desktop version, we connect to Lsa only as an untrusted caller. We do not attempt to exploit Tcb privilege or adjust the current
-        /// thread privilege to include Tcb.
-        /// </remarks>
-        public WindowsIdentity(string sUserPrincipalName)
+        public WindowsIdentity(string sUserPrincipalName, string type)
             : base(null, null, null, ClaimTypes.Name, ClaimTypes.GroupSid)
+        {
+            KerbS4ULogon(sUserPrincipalName, ref _safeTokenHandle);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the WindowsIdentity class for the user represented by the specified User Principal Name (UPN).
+        /// </summary>
+        public WindowsIdentity(string sUserPrincipalName)
+            : this(sUserPrincipalName, null)
+        { }
+
+        private static void KerbS4ULogon(string sUserPrincipalName, ref SafeAccessTokenHandle accessTokenHandle)
         {
             // Desktop compat: See comments below for why we don't validate sUserPrincipalName.
 
-            using (SafeLsaHandle lsaHandle = ConnectToLsa())
+            using (SafeLsaHandle lsaHandle = GetLsaHandle(s_sourceName))
             {
                 int packageId = LookupAuthenticationPackage(lsaHandle, Interop.SspiCli.AuthenticationPackageNames.MICROSOFT_KERBEROS_NAME_A);
```

Comment:
This is the only call to `LookupAuthenticationPackage`, which means that `Interop.SspiCli.AuthenticationPackageNames.MICROSOFT_KERBEROS_NAME_A` is the only argument ever provided, which means that every call to Ascii.GetBytes will produce the same value (in a different array).

Maybe this value should be pre-ASCII'd and held for the lifetime of the process?